modules/java-modules/hdfs/src/main/java/org/syslog_ng/hdfs/HdfsDestination.java

```diff
@@ -43,28 +43,110 @@
 import java.util.Map.Entry;
 import java.util.UUID;
 
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+import static java.lang.Math.abs;
+
 public class HdfsDestination extends StructuredLogDestination {
     private static final String LOG_TAG = "HDFS:";
     private static final String HADOOP_SECURITY_AUTH_KEY = "hadoop.security.authentication";
     private static final String DFS_SUPPORT_APPEND_KEY = "dfs.support.append";
 
-    HdfsOptions options;
-    Logger logger;
-
+    private HdfsOptions options;
+    private Logger logger;
     private FileSystem hdfs;
-
     private boolean isOpened;
     private int maxFileNameLength = 255;
     private boolean appendEnabled;
     private String archiveDir;
+    private HashMap<String, HdfsFile> openedFiles;
+    private Timer timeReap;
+    private Lock lock;
+
+    class TimeReapTask extends TimerTask {
+        private boolean closeFile(HdfsFile hdfsfile) {
+           logger.info("Closing file " + hdfsfile.getPath() + "; reason=reaping");
+
+           try {
+               hdfsfile.flush();
+               hdfsfile.close();
+           } catch (IOException e) {
+               printStackTrace(e);
+               return false;
+           }
+
+           if (archiveDir != null)
```

Comment:
I would not recommend to use this form (no braces).
Later it will be really hard to maintain this code (easy to make errors, just extend with one line and forget about the brace) and the general code style is against this: http://checkstyle.sourceforge.net/reports/google-java-style-20170228.html#s4.1-braces

Please fix this other places too (I can see it several places in the code)

Reply:
From `CONTRIBUTING.md`: 
/`We are using a coding style very similar to [GNU Coding Standards]`/
As far as I know there is no other style accepted. Therefore bringing up the `google-java-style` guide is bogus.
In fact we have a few examples not to use braces, even in java or C.

But as here I do not really have a preference I could change.

Reply:
I don't agree. almost all java coding standards say DO NOT use if without brace (google, oracle, others).
would be nice not to refactor this code to a worse format.